--- include/xen/interface/io/blkif.h.origin	2020-05-23 20:32:40.790540524 +0200
+++ include/xen/interface/io/blkif.h	2020-05-23 21:04:34.476891524 +0200
@@ -194,7 +194,7 @@
 struct blkif_request_rw {
 	uint8_t        nr_segments;  /* number of segments                   */
 	blkif_vdev_t   handle;       /* only for read/write requests         */
-#ifndef CONFIG_X86_32
+#if !defined(CONFIG_X86_32) && !defined(CONFIG_ARM)
 	uint32_t       _pad1;	     /* offsetof(blkif_request,u.rw.id) == 8 */
 #endif
 	uint64_t       id;           /* private guest value, echoed in resp  */
@@ -206,7 +206,7 @@
 	uint8_t        flag;         /* BLKIF_DISCARD_SECURE or zero.        */
 #define BLKIF_DISCARD_SECURE (1<<0)  /* ignored if discard-secure=0          */
 	blkif_vdev_t   _pad1;        /* only for read/write requests         */
-#ifndef CONFIG_X86_32
+#if !defined(CONFIG_X86_32) && !defined(CONFIG_ARM)
 	uint32_t       _pad2;        /* offsetof(blkif_req..,u.discard.id)==8*/
 #endif
 	uint64_t       id;           /* private guest value, echoed in resp  */
@@ -218,7 +218,7 @@
 struct blkif_request_other {
 	uint8_t      _pad1;
 	blkif_vdev_t _pad2;        /* only for read/write requests         */
-#ifndef CONFIG_X86_32
+#if !defined(CONFIG_X86_32) && !defined(CONFIG_ARM)
 	uint32_t     _pad3;        /* offsetof(blkif_req..,u.other.id)==8*/
 #endif
 	uint64_t     id;           /* private guest value, echoed in resp  */
@@ -227,7 +227,7 @@
 struct blkif_request_indirect {
 	uint8_t        indirect_op;
 	uint16_t       nr_segments;
-#ifndef CONFIG_X86_32
+#if !defined(CONFIG_X86_32) && !defined(CONFIG_ARM)
 	uint32_t       _pad1;        /* offsetof(blkif_...,u.indirect.id) == 8 */
 #endif
 	uint64_t       id;
@@ -235,7 +235,7 @@
 	blkif_vdev_t   handle;
 	uint16_t       _pad2;
 	grant_ref_t    indirect_grefs[BLKIF_MAX_INDIRECT_PAGES_PER_REQUEST];
-#ifndef CONFIG_X86_32
+#if !defined(CONFIG_X86_32) && !defined(CONFIG_ARM)
 	uint32_t      _pad3;         /* make it 64 byte aligned */
 #else
 	uint64_t      _pad3;         /* make it 64 byte aligned */
